mh/quest_utils.py

```diff
@@ -1548,921 +1745,63 @@ def read_quest_sm_data(fname):
-            'size': 0x00,
-            'hp_spread': 0x00,  # 0: fixed, 1: spread of 5, 2: spread of 3
-            'size_spread': 0x00
-        },
-        'monster_3': {
-            'type': Monster.none,
-            'boss_id': 0x0000,
-            'enabled': False,
-            'level': 0x00,  # 0x01 through 0x3c
-            'size': 0x00,
-            'hp_spread': 0x00,  # 0: fixed, 1: spread of 5, 2: spread of 3
-            'size_spread': 0x00
-        }
-    },
-    'objective_details': {
-        'main_quest': {
-            'type': 0x00000001,
-            'objective_type': Monster.deviljho,
-            'objective_num': 0x01,
-            'zenny_reward': 18000,
-            'hrp_reward': 1800,
-            'rewards_row_1': [
-                (ItemsType.timeworn_charm, 1, 33),
-                (ItemsType.deviljho_gem, 1, 4),
-                (ItemsType.deviljho_scalp, 1, 8),
-                (ItemsType.deviljho_hide, 1, 17),
-                (ItemsType.hvy_armor_sphere, 1, 11),
-                (ItemsType.shining_charm, 1, 19),
-                (ItemsType.deviljho_fang, 1, 8)
-            ],
-            'rewards_row_2': [
-                (ItemsType.deviljho_scalp, 1, 20),
-                (ItemsType.timeworn_charm, 1, 34),
-                (ItemsType.deviljho_fang, 1, 6),
-                (ItemsType.deviljho_hide, 1, 17),
-                (ItemsType.deviljho_gem, 1, 3),
-                (ItemsType.shining_charm, 1, 11),
-                (ItemsType.hvy_armor_sphere, 1, 9)
-            ],
-        },
-        'subquest_1': {
-            'description': "None",
-            'type': 0x00000000,
-            'objective_type': Monster.none,
-            'objective_num': 0x00,
-            'zenny_reward': 0,
-            'hrp_reward': 0x00000000,
-            'rewards_row_1': [],
-        },
-        'subquest_2': {
-            'description': "None",
-            'type': 0x00000000,
-            'objective_type': Monster.none,
-            'objective_num': 0x00,
-            'zenny_reward': 0,
-            'hrp_reward': 0x00000000,
-            'rewards_row_1': [],
-        },
-    },
-    'unknown': {
-        # 2 for large mon quest, 3 for small/delivery, 5 for jhen/ala
-        'unk_12': 0x00000002,
-        'unk_4': 0x00,
-        'unk_5': 0x00,
-        'unk_6': 0x00,
-        'unk_7': 0x00000000,
-        'unk_9': 0x00000000,
-        'unk_10': 0x00000000,
-        'unk_11': 0x00000000,
-    }
-}
+def byteify(input):
+    if isinstance(input, dict):
+        return {byteify(key): byteify(value)
+                for key, value in input.iteritems()}
+    elif isinstance(input, list):
+        return [byteify(element) for element in input]
+    elif isinstance(input, unicode):
+        return input.encode('utf-8')
+    else:
+        return input
+
+
+class QuestLoader:
+    def __init__(self, file_loc='event/quest_rotation.json'):
+        self.version = 0
+        self.version_incrementer = 28
```

Comment:
Is this an arbitrary constant?

Reply:
This gives us the ability to expand the quest rotation to up to 4 weeks (7 * 4 days) per update.